### ISSUE TITLE
[Bug Report]: Getting weird connectivity with Xbox Elite 2

### ISSUE THREAD

**Issue by mattminew** [ndeadly/MissionControl#494]
### Switch Firmware Version

14.1.1 (Latest)

### Atmosphère Version

1.3.2 (Latest)

### Mission Control Version

0.7.0 (Latest)

### Boot Method

Fusée

### Issue Description

I can only connect the Elite 2 controller to my switch by pairing it through change grip/order menu each time, as in it doesn't auto connect after pairing. My controller is downgraded to the correct firmware. 

I have a DS4 controller that behaves correctly as well so I'm not too sure what the issue is. Is anyone else getting this issue? Couldn't find much about this problem. 

Thanks!

**Comment by ndeadly** [comment]
That's strange. Never heard of anything like this happening before. Have you had it working previously on older firmwares to this?

Without knowing too much about what's actually going on, it sounds almost like the controller is successfully pairing but not being stored into the pairing database that the console uses to authenticate with previously paired devices. Try running my [btdb ](https://github.com/ndeadly/MissionControl/files/7227157/btdb-1300.zip) tool to see if the controller shows up in the database. It could also be worth clearing this database and re-pairing the controller in case something funky has happened to it.

**Comment by mattminew** [comment]
It's actually been going on for a while, even after a hard reset and reapplying atmosphere. Running btdb actually crashes my switch so that might be related.

**Comment by ndeadly** [comment]
> Running btdb actually crashes my switch so that might be related

Did you try the version I posted above? I haven't distributed the app via official channels in quite some time (I know places like the hb appstore still bundle it with the new releases for some reason), and an update was required to make it work on firmware 13.0.0 and above.

**Comment by mattminew** [comment]
Ah, I've been downloading mission control off of the hb appstore. The version you posted works, and I see the xbox elite wireless controller listed on the bluetooth devices, but the issues still there, it's not auto connecting.

**Comment by mattminew** [comment]
Sadly it still doesn't automatically reconnect, but that's ok now that I understand the problem. It's been going on for about a year so some conclusion is really relieving. Thanks for helping me understand though, you were a great help!

**Comment by ndeadly** [comment]
No problem. You can leave this issue open in case anyone else is having similar issues and might have additional insight. Unfortunately, not owning the controller myself means I can't really investigate much further.

I need to update my debugging tools at some point in the near future. I might post something here for you to try in the hope that I can catch something interesting in a log.

**Comment by mattminew** [comment]
Thanks, I appreciate it

**Comment by oaktrunks** [comment]
I believe I am having the same issue with my Elite 2 controller after downgrading it to 4.8.1908.0. Every time I turn the controller off and back on I have to pair it to the switch again. Let me know if there's anything I can do to add any additional insight.

**Comment by ndeadly** [comment]
You're the third person to report this issue now, so it definitely seems to more than an anomaly. In all three cases the controller has been downgraded from the BLE firmware. I'm curious whether it's related to the controller firmware, or whether something changed in the newer switch firmwares. If you are able to spin up a new throwaway emummc and downgrade it to 12.0.0 and/or 13.0.0, clear the database and re-pair the controller, I would be interested to know if that makes any difference to the behaviour.

**Comment by oaktrunks** [comment]
Okay, so I tried it on 13.0.0 and it still has the same issues. However once you clear the database the controller works properly until you plug it back into the computer. Then it starts acting up again until you clear the database and re-pair the controller. This holds true for 14.1.2 as well. I found this [thread](https://answers.microsoft.com/en-us/xbox/forum/all/xbox-elite-2-windows-10-wont-connect-after-using/fe07334e-60e0-4134-ad4c-b0a1b4b90bee) on the Microsoft support website, which indicates it's a controller problem, at least on old versions of the controller's firmware. 

Basically, any time you plug the controller into your PC and want to use it on the switch afterwards you have to wipe the controller database and re-pair and then it should work properly until the next time you plug it in.

**Comment by Masamune3210** [comment]
Maybe connecting it to a computer pairs it in some way to the computer? I know that the console/Xbox app on PC will try to associate a specific gamertag to a controller, but I'm pretty sure that's all either locally stored or server side and shouldn't affect Bluetooth pairing

**Comment by kakra** [comment]
> In all three cases the controller has been downgraded from the BLE firmware. I'm curious whether it's related to the controller firmware, or whether something changed in the newer switch firmwares

The 4.8 firmware of this controller is very unstable, it has all sorts of Bluetooth connection stability problems and depends a lot on quirks of the host Bluetooth stack.

**Comment by Masamune3210** [comment]
So its just really buggy controller firmware that happens to be the last one before they switched over to LE?
